parser.go

```diff
@@ -40,6 +40,57 @@ type Node[V comparable] struct {
 	Column int
 }
 
+// Left returns the left child in the case of a binary tree.
+func (p *Node[V]) Left() *Node[V] {
+	if len(p.Children) > 0 {
+		return p.Children[0]
+	}
+	return nil
+}
+
+// SetLeft sets the left child in the case of a binary tree and returns the
+// previous value.
+func (p *Node[V]) SetLeft(l *Node[V]) *Node[V] {
+	for len(p.Children) < 1 {
+		p.Children = append(p.Children, nil)
+	}
+	old := p.Children[0]
+	p.Children[0] = l
+	l.Parent = p
+	return old
+}
+
+// Right returns the right child in the case of a binary tree.
+func (p *Node[V]) Right() *Node[V] {
+	if len(p.Children) > 1 {
+		return p.Children[1]
+	}
+	return nil
+}
+
+// SetRight sets the right child in the case of a binary tree and returns the
+// previous value.
+func (p *Node[V]) SetRight(r *Node[V]) *Node[V] {
+	for len(p.Children) < 2 {
+		p.Children = append(p.Children, nil)
+	}
+	old := p.Children[1]
+	p.Children[1] = r
+	r.Parent = p
+	return old
+}
+
+// ReplaceChild replaces the first given node with another node. The inserted
+// node's parent is updated. The removed node's parent is not updated.
```

Comment:
Comment that `l`'s children are not moved and do not become `r`'s children.

Reply:
I just went ahead and removed this since it's not a particularly common operation and, as you noted, it's ambiguous. We also have the `Parser.Replace` method which is better anyway.